canonical/ubuntu.com
# Add study labs

## Done

1. Provide Study Labs product listing for the "Prepare" button.
2. How do we provide the Study Lab product?
All data available for prepare in `study_labs_listing`

3. Include the URL to the Study Lab. 
Added: https://qa.cube.ubuntu.com/courses/course-v1:CUBE+study_labs+2020/course/

4. How do we know if the Study Lab is purchased? 
`has_study_labs` will return `null` if there are no purchased study_labs

## QA
- Visit https://ubuntu-com-10714.demos.haus/cube/microcerts.json?test_backend=true
- Check that it's returning all the data, plus `study_labs_listing`, which includes the name of the study lab, its price, `product_listing_id`, `has_study_labs` etc... all the data needed to purchase
- If you haven't got a study lab, make a purchase POST https://ubuntu-com-10714.demos.haus/cube/microcerts/purchase.json using this new product_id
```
{
   "account_id": <your-account-id>
   "product_listing_id": <study-lab-product-id>, # you can get this from the `microcerts.json` endpoint
   "preview": "true"
}
```
- Do the complete purchase
```
{
   "account_id": <your-account-id>
   "product_listing_id": <study-lab-product-id>, # you can get this from the `microcerts.json` endpoint
   "preview": "false"
}
```
- After you have purchased a study_lab, go to https://ubuntu-com-10714.demos.haus/cube/microcerts.json and the `has_study_labs` variable should be set to `true`.
- If the former is `true` you should be able to access the study labs, by navigating to the URL in the microcerts.json `study_labs_url` variable.

## Issue / Card

Fixes https://github.com/canonical-web-and-design/ubuntu.com/issues/10713
Fixes https://sentry.is.canonical.com/canonical/ubuntu-com/issues/22727/activity/
